utils/extra.py

```
@@ -216,49 +218,42 @@ def linecount():
 
 
 # will require a better name and variables down below
-class RtfmObject(NamedTuple):
+class RtfmObject(typing.NamedTuple):
     name: str
     url: str
 
     def __str__(self) -> str:
         return self.name
 
 
-async def rtfm(bot, url):
-
+async def rtfm(bot, url) -> list[RtfmObject]:
     # wip
-    response = await bot.session.get(f"{url}objects.inv")
-    content = await response.read()
-
-    lines = content.split(b"\n")
-
-    first_10_lines = lines[:10]
-
-    first_10_lines = [n for n in first_10_lines if not n.startswith(b"#")]
-
-    lines = first_10_lines + lines[10:]
-    joined_lines = b"\n".join(lines)
-    full_data = zlib.decompress(joined_lines)
-    normal_data = full_data.decode()
-    new_list = normal_data.split("\n")
-
-    results = []
-    for x in new_list:
-        try:
-            name, type, _, fragment, *label = x.split(" ")
-
-            text = " ".join(label)
-
-            if text != "-":
-                label = text
-
-            else:
-                label = name
-
-        except:
-            continue
-
-        fragment = fragment.replace("$", name)
-        results.append(RtfmObject(label, url + fragment))
-
+    async with bot.session.get(f"{url}objects.inv") as response:
+        content: bytes = await response.read()
+        lines: list[bytes] = content.split(b"\n")
+        
+        cleared_out = [n for n in lines[:10] if not n.startswith(b"#")]
+
+        lines = cleared_out + lines[10:]
+        data = b"\n".join(lines)
+        
+        data = zlib.decompress(data)
+        data = data.decode()
+        data = data.split("\n")
+
+        results = []
+        for x in data:
+            try:
+                name, type, _, fragment, *label = x.split(" ")
+                text = " ".join(label)
+
+                if text != "-":
+                    label = text
+                else:
+                    label = name
+            except:
```

consider a non bare except block

---

Solved.